sw/airborne/firmwares/rotorcraft/guidance/guidance_h.c

```diff
@@ -404,8 +404,16 @@ void guidance_h_nav_enter(void)
   /* horizontal position setpoint from navigation/flightplan */
   guidance_h_set_pos(nav.carrot.y, nav.carrot.x);
   reset_guidance_reference_from_current_position();
+
+#if HYBRID_NAVIGATION
+  /*Obtain eulers with zxy rotation order*/
+  struct FloatEulers eulers_zxy;
+  float_eulers_of_quat_zxy(&eulers_zxy, stateGetNedToBodyQuat_f());
+  nav_heading = eulers_zxy.psi;
+#else
```

Comment:
This `HYBRID_NAVIGATION` flag should have been removed from the code in latest master. I'd rather not adding it back.
Since this function ends with a call `guidance_h_run_enter` which is implemented by the underlying guidance, you can reset the nav heading with the proper value in the function `guidance_indi_enter`.
Also, I'm wondering if this is really necessary, since the the rotations starts around the z axis in both cases, it should correspond to the same psi angle, or am I wrong ?

Reply:
Thanks for your comment, will change! Unfortunately one of our drones is flying on an old branch, as the AP crashes on the latest master due to something happening in the control allocation, which I still have to figure out.

Unfortunately it is necessary, it is true that the first rotation is around Z, but the first rotation has to be different in both cases. Imagine a pitch up of 91 degrees in ZXY. With ZYX you need 180 degrees psi, 89 degrees pitch up and 180 degrees roll.